app/src/main/java/fr/free/nrw/commons/utils/SystemThemeUtils.java

```diff
@@ -0,0 +1,49 @@
+package fr.free.nrw.commons.utils;
+
+import android.content.Context;
+import android.content.res.Configuration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.settings.Prefs;
+
+public class SystemThemeUtils {
+
+    private Context context;
+    private JsonKvStore applicationKvStore;
+
+    @Inject
+    public SystemThemeUtils(Context context, @Named("default_preferences") JsonKvStore applicationKvStore) {
+        this.context = context;
+        this.applicationKvStore = applicationKvStore;
+    }
+
+    // Return true is system wide dark theme is enabled else false
+    public boolean getSystemDefaultThemeBool(String theme) {
+        if (context.getResources().getString(R.string.theme_dark_name).equals(theme)) {
```

Comment:
context has a method `getString()`, you don't need resources to get a String

Reply:
Thanks, changes made!